pkg/config/config.go

```diff
@@ -1169,13 +1169,26 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 		applyOverrideFuncs(config)
 	}()
 
-	if err := config.ReadInConfig(); err != nil {
-		if errors.Is(err, os.ErrPermission) {
-			log.Warnf("Error loading config: %v (check config file permissions for dd-agent user)", err)
+	for _, confFilePath := range confFilePaths {
```

Comment:
I think you might want to check if `len(confFilePath) > 0`, since it is an empty string by default (at least in the system-probe).